internal/engine/engine.go

```diff
@@ -620,6 +620,23 @@ func (engine *Engine) getRolePolicySets(ctx context.Context, scope string, roles
 	}
 
 	sets, err := engine.policyLoader.GetAll(ctx, roleModIDs)
+	if err == nil {
+		// compile time check against colliding scopePermission settings in shared scope
+		var requireParentalConsent, overrideParent int
+		for _, r := range sets {
+			switch r.GetRolePolicy().ScopePermissions { //nolint:exhaustive
+			case policyv1.ScopePermissions_SCOPE_PERMISSIONS_REQUIRE_PARENTAL_CONSENT_FOR_ALLOWS:
+				requireParentalConsent++
+			case policyv1.ScopePermissions_SCOPE_PERMISSIONS_OVERRIDE_PARENT:
+				overrideParent++
+			}
+
+			if requireParentalConsent > 0 && overrideParent > 0 {
+				err = errors.New("invalid scope permissions: role policies cannot combine different scope permissions within the same scope")
```

Comment:
Couldn't we actually do this at compile time? If we don't allow role policies to have different permission scopes, does it make sense to allow them to exist in the repo, hoping that they don't ever get used at the same time?

Reply:
I just realised that we do need to keep this check here for dynamic stores like databases where policies could be added at any time. However, we should still do the same check at index build time as well so that `cerbos compile` command can warn users when they have an invalid repo.

Reply:
I actually tried a couple of things here:

Compile time didn't seem to work because `GetAll` is (understandably) a policy-agnostic API. It returns a list of individual policy sets, rather than a single set with all role policies. The most immediate logical boundary with access to the exclusive, related set of role policies was here. It's also interface implementation agnostic.

I also considered implementing at index build time, but I was nervous to implement there because I'm not sure we can guarantee that that code path might always be used?

Reply:
If we have a disk-based store that contains role policies, I think we should check that they all share the same `scopePermissions` value. Otherwise, it's a bit of a gotcha because a chance combination of roles can result in an error at runtime.

That could be done in a different PR though. Doesn't have to be in this one.

Reply:
Oh, I hadn't refreshed so hadn't seen your follow up.

I can add back in the index build time check though (stupidly _just_ deleted the git stash I had with that change 🤠. Need a longer TTL)